controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -345,6 +345,44 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 		return err
 	}
 
+	// Watch for changes to secondary resource Deployment and requeue the owner Observatorium
```

Comment:
don't understand the watch function here fully.
The purpose for the watch is to update the thanosReplica? why need to trigger reconcile?

Reply:
the watch just update the `thanosReplicas` in memory. still reply on the reconcile to update the observatorium CR. normally, we just need to watch for update. watch create is for deleting the mco-o case.